common-messaging/src/main/scala/org/genivi/sota/messaging/MessageBusManager.scala

```diff
@@ -104,6 +104,23 @@ object MessageBusManager {
         Xor.right(MessageBusPublisher(system.eventStream.publish))
       case _ => Xor.left(new Missing("Unknown messaging mode specified"))
     }
+  }
+
+  def getDeviceDeletedPublisher(system: ActorSystem, config: Config)
+  : ConfigException Xor MessageBusPublisher[DeviceDeleted] = {
+    val log = Logging.getLogger(system, this.getClass)
 
+    config.getString("messaging.mode") match {
+      case "nats" =>
+        log.info("Starting messaging mode: NATS")
+        NatsClient.getDeviceDeletedPublisher(system, config)
+      case "kinesis" =>
+        log.info("Starting messaging mode: Kinesis")
+        KinesisClient.getDeviceDeletedPublisher(system, config)
+      case "test" =>
+        log.info("Starting messaging mode: Test")
+        Xor.right(MessageBusPublisher(system.eventStream.publish))
+      case _ => Xor.left(new Missing("Unknown messaging mode specified"))
+    }
   }
 }
```

Comment:
This can be abstracted further. ie, we don't need a method for each message type, we can use a type parameter. For example, to implement a generic method in the nats client:

```
  def getPublisher[T <: Message](system: ActorSystem, config: Config): ConfigException Xor MessageBusPublisher[T] = {
    getClient(system, config).map { conn =>
      MessageBusPublisher { msg =>
        conn.publish(msg.streamName, msg.asJson.noSpaces)
      }
    }
  }
```

And we can do the same for kinesis, and then we can have a `getPublisher[T <: Message]` method in the MessageBusManager as well. 

Also the same for subscriptions